x-pack/plugins/cases/server/services/attachments/index.ts

```diff
@@ -364,7 +411,9 @@ export class AttachmentService {
            */
           references:
             refsWithExternalRefId.length > 0 || didDeleteOperation
-              ? refsWithExternalRefId
+              ? uniqBy([...refsWithExternalRefId, ...extractedReferences], 'id')
```

Comment:
`uniqBy` is needed because when you create the attachment the `extract` function will return the same references as here. It is possible the `extractedReferences` to contain some of the references in `refsWithExternalRefId`.